### PR TITLE
Add README documentation explaining that the keychain sharing entitlement is required

### LINKED ISSUE #213: Migrating Objects from Pre Catalina unexpected results
Hi there! I have a macOS 10.15 non-sanboxed, hardened-runtime app _without_ the Keychain Sharing entitlement enabled.

I have a very simple valet setup

```swift
let valet = Valet.valet(with: Identifier(nonEmpty: Bundle.main.bundleIdentifier!)!, accessibility: .whenUnlocked)
```

I have recently upgraded to Valet 3.2.8. Please note this issue below did not occur before the ```kSecUseDataProtectionKeychain = true``` update.

On 10.14 and before, writing to this keychain puts a "application password" in the "login" keychain.

In Catalina, however, writing to this keychain results in a "Missing Entitlement" error. I know this isn't a Valet-specific issue. I go ahead and add the Keychain Sharing entitlement as I read somewhere that that could fix the issue - lo and behold I'm able to write to the keychain! 

However, It's not writing to the "login" keychain but instead the "iCloud" keychain and seems to be using the default Keychain Access Group identifier ($teamID).($appIdentifier) as the access group instead of granting the app access directly. This is all a bit weird, and also ```migrateObjectsFromPreCatalina()``` does not perform an in-place adjustment but instead creates a duplicate keychain item because it now is an "iCloud" item and not a "login" item.

Any insight here is much appreciated!

### PR DESCRIPTION
**Pull request by dfed**
This PR should resolve #213.

cc @gregcotten

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -171,7 +171,9 @@ However, because the Keychain is effectively disk storage, there is no guarantee
...
 
-### Supporting macOS Catalina (10.15) and later
+### Integrating Valet into a macOS application
+
+Your macOS application must have the Keychain Sharing entitlement in order to use Valet, even if your application does not intend to share keychain data between applications. For instructions on how to add a Keychain Sharing entitlement to your application, read [Apple's documentation on the subject](https://developer.apple.com/documentation/security/keychain_services/keychain_items/sharing_access_to_keychain_items_among_a_collection_of_apps). The subsection on "Add Apps to One or More Keychain Access Groups" is particularly helpful, as it includes a screenshot showing Xcode's Signing & Capabilities screen. For more information on why this requirement exists, see [issue #213](https://github.com/square/Valet/issues/213).
```
Comment (fdiaz) [inline README.md]: You can use https://developer.apple.com/documentation/security/keychain_services/keychain_items/sharing_access_to_keychain_items_among_a_collection_of_apps#2974918 to link directly to the "Add Apps to One or More Keychain Access Groups"` section

Comment (dfed) [reply README.md]: That's way better! When you have a chance, I'd love to know how you generated that link. I tried to find link anchors on that page but came up short.

Comment (fdiaz) [reply README.md]: I just inspected the HTML 😁 


image

Comment (dfed) [reply README.md]: Smart!! I'm going to have to steal that trick.

**Pull request merged**